File: app/src/main/java/com/example/bmiapp/ResultActivity.java
package com.example.bmiapp;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class ResultActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result);
        float heightValue = getIntent().getFloatExtra("h", 0);
        float weightValue = getIntent().getFloatExtra("w", 0);
        float age = getIntent().getIntExtra("age", 0);
        boolean isMale = getIntent().getBooleanExtra("isMale", true);

        BMIHelper helper = new BMIHelper();
        float bmi = helper.calculateBMI(heightValue, weightValue);

        CalorieCalculator calculator = new CalorieCalculator();
        double calories = calculator.calculateCalories(weightValue, heightValue, age, isMale);

        TextView bmiTxt = findViewById(R.id.bmiTxt);
        bmiTxt.setText(bmi + "");

        TextView caloriesTxt = findViewById(R.id.caloriesTxt);
        caloriesTxt.setText(calories + "");

        FeedSuggestionGenerator generator = new FeedSuggestionGenerator();
        String feed = generator.generateFeed(bmi);

        TextView feedTxt = findViewById(R.id.feedTxt);
        feedTxt.setText(feed);
    }
}

File: app/src/main/java/com/example/myapplication/ResultActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class ResultActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_result)
        val heightValue = intent.getFloatExtra("h", 0f)
        val weightValue = intent.getFloatExtra("w", 0f)
        val age = intent.getIntExtra("age", 0).toFloat()
        val isMale = intent.getBooleanExtra("isMale", true)

        val helper = BMIHelper()
        val bmi = helper.calculateBMI(heightValue, weightValue)

        val calculator = CalorieCalculator()
        val calories = calculator.calculateCalories(weightValue, heightValue, age, isMale)

        val bmiTxt = findViewById<TextView>(R.id.bmiTxt)
        bmiTxt.text = bmi.toString() + ""

        val caloriesTxt = findViewById<TextView>(R.id.caloriesTxt)
        caloriesTxt.text = calories.toString() + ""

        val generator = FeedSuggestionGenerator()
        val feed = generator.generateFeed(bmi)

        val feedTxt = findViewById<TextView>(R.id.feedTxt)
        feedTxt.text = feed
    }
}
